freqtrade/rpc/api_server/api_schemas.py

```diff
@@ -355,7 +360,9 @@ class PairHistory(BaseModel):
 
     class Config:
         json_encoders = {
-            datetime: lambda v: v.strftime(DATETIME_PRINT_FORMAT),
+            datetime: lambda v: v.strftime(DATETIME_PRINT_FORMAT)
+            if dtype_string(v) in ["timedelta64", "datetime64", "datetime", "Timestamp"]
```

Comment:
Did you actually look at which field/row is "NaT"?  (maybe you can provide a simple strategy that causes this?)

this is a pretty broad approach - and i'm not sure what you need to do to a dataframe to get a "NaT" entry in your date column (it almost certainly means your strategy is messing with the dataframe - like adding lines - which means the strategy will probably raise errors in live.

Reply:
I think the correct place to fix this would be in `rpc.py` - where we also replace `np.NaN` with None.

That said - it appears like currently it's not possible to convert pd.NaT to "anything" - so i've opened an issue against pandas (https://github.com/pandas-dev/pandas/issues/45836).
I'd want to wait on the outcome of that issue before deciding if/how we implement a fix.

Reply:
I think that is for sure the sensible approach, but so far it's not even sure *if* they will fix that (glimpsing at your discussions of there)

And for now this part would still be broken, even though we have a "fix" and another one (fixing it in `rpc.py`) lined up already.

How about we implement want of this bandaid fixes for now, reference in the comments this PR and also write a unit test with the strategy @Bloodhunter4rc provided as a litmus test?

Once pandas fixes it proper, we can just switch over do it then

Reply:
so far, we've barely given pandas time to actually respond (the issue is barely 2 days old).

if any, i'd use the approach in `rpc.py`. Updating the formatter (as you're proposing here) will only fix one particular issue - but may still fail should a column contain NaN / inf.

I don't think we'll need the strategy for that though - as we can construct a fake dataframe that proves the "failure" without issues.

Reply:
:+1: 

okay, I will look into the `rpc.py` thing, worst case I learn something.

even better if the strategy is not needed, just wanted to make sure that there is a test in place to verify the proper fix once the band-aided would needed to be ripped of.

what are the properties of the fake dataframe to force the "failure"? just any datetime colume containing NaT/NaN/inf?

Reply:
a datetime-column with `pd.NAT`. (maybe also with some valid dates).

Also, maybe a numeric column which includes inf/-inf or `pd.np.NaN` ... as i think the same bug is there as well (that's why i say - the current fix doesn't fix all instances of the problem).